### PR TITLE
Fix page errors caused when volunteer form re-renders due to update error

### LINKED ISSUE #708: Better error when trying to change an email to an email already in the system
ERROR in prod https://rubyforgood.slack.com/archives/C010Z8RK8E6/p1599610264099800
Sarah-admin tried to assign sara-supervisor email to the same as sarah-admin’s email address

### PR DESCRIPTION
**Pull request by littleforest**
### What github issue is this PR for, if any?
Resolves #708 

### What changed, and why?
When the volunteer form had to re-render due to an update error, the instance variables defined in the `edit` method were not available, causing errors. Instead of re-defining the instance variables in the `update` method, I moved the instance variables to methods.

### How is this tested? (please write tests!) 💖💪
Added request and system specs to test the case when a non-unique email address is used to update the form.

### Screenshots please :)

Screen Shot 2020-09-11 at 1 18 27 PM

### Notes
Will create follow-up task to prettify the error message.

### CODE REVIEW

**Comment by compwron** [comment]
Very nice! 
I'm going to hold off on approve/merge for this for a few hours in the hopes of letting #624 get sorted out, but I'll merge everything tomorrow saturday morning before r4g starts up at the latest :)

File: app/controllers/volunteers_controller.rb [hunk 1]
```diff
@@ -20,8 +20,7 @@ def create
   end
 
   def edit
-    @case_assignments = @volunteer.case_assignments.includes(:casa_case)
-    @available_casa_cases = _get_available_casa_cases
+    # TODO do we need something here?
```
Comment (littleforest) [inline app/controllers/volunteers_controller.rb:23]: Don't need anything here currently.

**Comment by littleforest** [comment]
@compwron can you wait to merge? Wanted to double check some of the other changes in the controller first.

**Comment by compwron** [comment]
@littleforest yep I will wait to merge :)

**Comment by littleforest** [comment]
@compwron ready for re-review!

**Review by jmkoni — APPROVED** [review]

**Pull request merged**